glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/PingUploadWorker.kt

```diff
@@ -173,9 +85,33 @@ class PingUploadWorker(context: Context, params: WorkerParameters) : Worker(cont
          * @return true if process was successful
          */
         internal fun uploadPings(): Boolean {
-            val storageDirectory = File(Glean.getDataDir(), PINGS_DIR)
+            var ffiTask = LibGleanFFI.INSTANCE.glean_get_upload_task()
+            var task = ffiTask.toPingUploadTask()
+            while (task is PingUploadTask.Upload) {
+                // Get the request
+                val request = task.request()!!
+                // Upload the request
+                // If the status is `null` there was some kind of unrecoverable error
+                // so we return 400 which will ensure this gets treated as such.
+                @Suppress("MagicNumber")
```

Comment:
Can we define a constant so that we don't need to suppress the "Magic Number"?  This might also help to document why you are replacing the null with an error status.